pkg/archive/changes.go

```diff
@@ -359,7 +359,7 @@ func ChangesSize(newDir string, changes []Change) int64 {
 }
 
 // ExportChanges produces an Archive from the provided changes, relative to dir.
-func ExportChanges(dir string, changes []Change) (Archive, error) {
+func ExportChanges(dir string, changes []Change, excludes []string) (Archive, error) {
```

Comment:
Sounds like what I expected. Perhaps just `exportChanges` would suffice,
since it won't collide with the exported function and is clearly related.
The `Impl` suffix feels a little java-ish. :-)
On May 27, 2015 01:19, "Shijiang Wei" <notifications@github.com> wrote:

> In pkg/archive/changes.go
> <https://github.com/docker/docker/pull/12594#discussion_r31103948>:
>
> > @@ -359,7 +359,7 @@ func ChangesSize(newDir string, changes []Change) int64 {
> >  }
> >
> >  // ExportChanges produces an Archive from the provided changes, relative to dir.
> > -func ExportChanges(dir string, changes []Change) (Archive, error) {
> > +func ExportChanges(dir string, changes []Change, excludes []string) (Archive, error) {
>
> @vbatts <https://github.com/vbatts> If add a new function I thinks I'll
> do it like this:
> 1. Rename the current ExportChanges to exportChangesImpl.
> 2. Call exportChangesImpl in ExportChanges(dir string, changes []Change)
> with an empty string slice.
> 3. Add a new function ExportChanges(dir string, changes []Change,
> excludes []string) to call exportChangesImpl directly.
>
> Please let me know if it makes sense. Thanks.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/docker/docker/pull/12594/files#r31103948>.
>